samples/Hardware.Esp32.Rmt/InfraredRemoteReceiver/InfraredRemoteReceiverSample/Decoder.cs

```diff
@@ -0,0 +1,157 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Text;
+using nanoFramework.Hardware.Esp32.Rmt;
+
+namespace InfraredRemote
```

Comment:
Well that was my first idea. I based this on some Arduino library, though I made it working fully for Nec protocol only with some shortcuts.

I tried with sending other protocols signals from my phone but either receiver is poor quality or sender. With real LG TV remote it worked way better on Nec protocol. 

As MrCSharp mentioned he is working on another decoder, so it might be useful to include those two as examples in separate binding and let community add another one. Then of course as he also stated in comments here some refactoring would need to be applied.

Any idea for the name of the binding?

Reply:
`SignalEncoderDecoder` maybe?

Reply:
Hmm yes and no, as always :D OregonScientific looks similar but I also have an example: water meters reading and then decoding: https://github.com/weetmuts/wmbusmeters - huge and complex.

Looks familiar right? But following you on this, we would have growing lib that should support another protocol/encoding/decoding that probably looks similar but will be used just by one transmission medium. I wouldn't like to go that way.

That is also the reason that finally I wanted to add a sample instead of library.

Reply:
OK, so we can start with this sample and then let's see